sanitycheck samples/video/capture/sample.video.capture fails to build on mimxrt1064_evk

`scripts/sanitycheck -s samples/video/capture/sample.video.capture -p mimxrt1064_evk`

Gets:
```
Scanning dependencies of target drivers__video
[ 86%] Building C object zephyr/drivers/video/CMakeFiles/drivers__video.dir/video_common.c.obj
/home/galak/git/zephyr/drivers/video/video_common.c:20:8: error: unknown type name 'mem_block'
   20 | static mem_block video_block[CONFIG_VIDEO_BUFFER_POOL_NUM_MAX];
      |        ^~~~~~~~~
/home/galak/git/zephyr/drivers/video/video_common.c: In function 'video_buffer_alloc':
/home/galak/git/zephyr/drivers/video/video_common.c:32:10: error: assignment to 'struct k_mem_block *' from incompatible pointer type 'int *' [-Werror=incompatible-pointer-types]
   32 |    block = &video_block[i];
      |          ^
/home/galak/git/zephyr/drivers/video/video_common.c: In function 'video_buffer_release':
/home/galak/git/zephyr/drivers/video/video_common.c:61:21: error: request for member 'data' in something not a structure or union
   61 |   if (video_block[i].data == vbuf->buffer) {
      |                     ^
/home/galak/git/zephyr/drivers/video/video_common.c:62:10: error: assignment to 'struct mem_block *' from incompatible pointer type 'int *' [-Werror=incompatible-pointer-types]
   62 |    block = &video_block[i];
      |          ^
cc1: all warnings being treated as errors
gmake[2]: *** [zephyr/drivers/video/CMakeFiles/drivers__video.dir/build.make:82: zephyr/drivers/video/CMakeFiles/drivers__video.dir/video_common.c.obj] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:3217: zephyr/drivers/video/CMakeFiles/drivers__video.dir/all] Error 2

```

--- a/drivers/video/video_common.c
+++ b/drivers/video/video_common.c
@@ -17,12 +17,12 @@ struct mem_block {
 	void *data;
 };
 
-static mem_block video_block[CONFIG_VIDEO_BUFFER_POOL_NUM_MAX];
+static struct mem_block video_block[CONFIG_VIDEO_BUFFER_POOL_NUM_MAX];
 
 struct video_buffer *video_buffer_alloc(size_t size)
 {
 	struct video_buffer *vbuf = NULL;
-	struct k_mem_block *block;
+	struct mem_block *block;
 	int i;
 
 	/* find available video buffer */
@@ -57,7 +57,7 @@ void video_buffer_release(struct video_buffer *vbuf)
 	int i;
 
 	/* vbuf to block */
-	for (i = 0; i < ARRAY_SIZE(video_buf); i++) {
+	for (i = 0; i < ARRAY_SIZE(video_block); i++) {
 		if (video_block[i].data == vbuf->buffer) {
 			block = &video_block[i];
 			break;

